iceoryx_utils/test/moduletests/test_semaphore_module.cpp

```diff
@@ -42,216 +98,257 @@ class Semaphore_test : public Test
     }
 };
 
-TEST_F(Semaphore_test, CreateNamedSemaphore)
+INSTANTIATE_TEST_CASE_P(Semaphore_test, Semaphore_test, Values(&CreateNamedSemaphore, &CreateUnnamedSemaphore));
+
+TEST_F(SemaphoreCreate_test, CreateNamedSemaphore)
 {
     auto semaphore = iox::posix::Semaphore::create("/fuuSem", S_IRUSR | S_IWUSR, 10);
     EXPECT_THAT(semaphore.has_error(), Eq(false));
 }
 
-TEST_F(Semaphore_test, CreateExistingNamedSemaphore)
+TEST_F(SemaphoreCreate_test, CreateExistingNamedSemaphore)
 {
     auto semaphore = iox::posix::Semaphore::create("/fuuSem1", S_IRUSR | S_IWUSR, 10);
     auto semaphore2 = iox::posix::Semaphore::create("/fuuSem1", S_IRUSR | S_IWUSR, 10);
     ASSERT_EQ(semaphore.has_error(), false);
     ASSERT_EQ(semaphore2.has_error(), true);
 }
 
-TEST_F(Semaphore_test, CreateLocalUnnamedSemaphore)
+TEST_F(SemaphoreCreate_test, CreateLocalUnnamedSemaphore)
 {
     auto semaphore = iox::posix::Semaphore::create(10);
     EXPECT_THAT(semaphore.has_error(), Eq(false));
 }
 
-TEST_F(Semaphore_test, OpenNamedSemaphore)
+TEST_F(SemaphoreCreate_test, OpenNamedSemaphore)
 {
     auto semaphore = iox::posix::Semaphore::create("/fuuSem", S_IRUSR | S_IWUSR, 10);
     auto semaphore2 = iox::posix::Semaphore::create("/fuuSem", S_IRUSR | S_IWUSR);
     EXPECT_THAT(semaphore.has_error(), Eq(false));
     EXPECT_THAT(semaphore2.has_error(), Eq(false));
 }
 
-TEST_F(Semaphore_test, OpenNonExistingNamedSemaphore)
+TEST_F(SemaphoreCreate_test, OpenNonExistingNamedSemaphore)
 {
     auto semaphore2 = iox::posix::Semaphore::create("/fuuSem", S_IRUSR | S_IWUSR);
     EXPECT_THAT(semaphore2.has_error(), Eq(true));
 }
 
-TEST_F(Semaphore_test, GetValueValidUnnamedSemaphore)
+TEST_P(Semaphore_test, PostIncreasesSemaphoreValue)
 {
-    auto semaphore = iox::posix::Semaphore::create(1234);
-    int i;
-    EXPECT_THAT(semaphore->getValue(i), Eq(true));
-    EXPECT_THAT(i, Eq(1234));
-}
+    for (int i = 0; i < 12; ++i)
+        sut->post();
 
-TEST_F(Semaphore_test, GetValueValidNamedSemaphore)
-{
-    auto semaphore = iox::posix::Semaphore::create("/fuuSem", S_IRUSR | S_IWUSR, 4337);
-    int i;
-    EXPECT_THAT(semaphore->getValue(i), Eq(true));
-    EXPECT_THAT(i, Eq(4337));
+    int value;
+    ASSERT_THAT(sut->getValue(value), Eq(true));
+    EXPECT_THAT(value, Eq(12));
 }
 
-TEST_F(Semaphore_test, PostValidUnnamedSemaphore)
+TEST_P(Semaphore_test, WaitDecreasesSemaphoreValue)
 {
-    auto semaphore = iox::posix::Semaphore::create(1234);
-    EXPECT_THAT(semaphore->post(), Eq(true));
-    EXPECT_THAT(semaphore->post(), Eq(true));
-    EXPECT_THAT(semaphore->post(), Eq(true));
-
-    int i;
-    EXPECT_THAT(semaphore->getValue(i), Eq(true));
-    EXPECT_THAT(i, Eq(1237));
+    for (int i = 0; i < 18; ++i)
+        sut->post();
```

Comment:
it's so tempting ;)

Reply:
Please, go on? What do you mean ??

Reply:
b ... r ... a ... c ... e ... s

Reply:
but in the tests I would be fine without braces ;)

Reply:
braces added